### PR TITLE
fix: dont spam rpc server

### PR DESCRIPTION
**Pull request by samsiegart**
refs https://github.com/Agoric/agoric-sdk/issues/6870

### CODE REVIEW

File: src/lib/rpc.ts [hunk 1]
```diff
@@ -102,11 +103,18 @@ export const makeRpcUtils = async ({ agoricNet }) => {
       ),
   };
 
+  const storageWatcher = makeAgoricChainStorageWatcher(
+    rpcAddrs[Math.floor(Math.random() * rpcAddrs.length)],
```
Comment (turadg) [inline src/lib/rpc.ts:107]: https://lodash.com/docs/4.17.15#sample would convey this well.
```suggestion
    _.sample(rpcAddrs),
```

if we have tree shaking then adopting https://www.npmjs.com/package/lodash-es would have only a small bundle weight

File: src/lib/wallet.ts [hunk 2]
```diff
@@ -443,24 +444,22 @@ export const usePublishedKeys = (path: string) => {
...
-      setData(publishedValue.value);
-      setStatus(LoadStatus.Received);
-    };
-    fetchData().catch(e => console.error('useEffect error', e));
-  }, [path, walletUtils]);
+    const { storageWatcher } = rpcUtils;
+
+    setStatus(LoadStatus.Waiting);
+
+    return storageWatcher.watchLatest(
```
Comment (turadg) [inline src/lib/wallet.ts:454]: much more readable.

imo the `XXX cleanup?` comment is done

**Review by turadg — APPROVED** [review]

File: src/lib/wallet.ts (hunk 2 above)
Comment (samsiegart) [reply src/lib/wallet.ts:452]: Oh true, nice

File: src/lib/rpc.ts (hunk 1 above)
Comment (samsiegart) [reply src/lib/rpc.ts]: Done. It added 7KB uncompressed / 2KB compressed, not terrible I guess.

**Pull request merged**